exts/ckanext-catalogdatagov/ckanext/catalogdatagov/logic/action/update.py

```diff
@@ -0,0 +1,434 @@
+import datetime
+import json
+import logging
+from sqlalchemy import exc
+
+from ckanext.harvest.logic.action.update import (
+    _make_scheduled_jobs, harvest_job_list, HarvestJob,
+    HarvestObject, resubmit_jobs, resubmit_objects, and_,
+    toolkit, get_action, config, get_gather_publisher,
+    PackageSearchIndex, harvest_job_dictize, logic, mailer)
+from ckan.logic import chained_action
+from ckan.plugins.toolkit import check_access
+from ckan.lib import search
+import ckan.lib.mailer
+
+
+log = logging.getLogger(__name__)
+
+
+@chained_action
+def harvest_jobs_run(up_func, context, data_dict):
+    log.info('Harvest job run: %r', data_dict)
+    check_access('harvest_jobs_run', context, data_dict)
+
+    model = context['model']
+    session = context['session']
+
+    source_id = data_dict.get('source_id')
+
+    if not source_id:
+        _make_scheduled_jobs(context, data_dict)
+
+    context['return_objects'] = False
+
+    # Not sure if this is needed
+    # set_harvest_system_info(
+    #     context, 'last_run_time', datetime.datetime.utcnow())
+
+    # Flag finished jobs as such
+    jobs = harvest_job_list(
+        context, {'source_id': source_id, 'status': u'Running'})
+
+    if len(jobs):
+        log.info('Job found')        
+
+        package_index = PackageSearchIndex()
+
+        for job in jobs:
+            log.info('Job info {}'.format(job))
+            if job['gather_finished']:
+                log.info('Job finished')
+                objects = session.query(HarvestObject.id) \
+                    .filter(HarvestObject.harvest_job_id == job['id']) \
+                    .filter(and_(
+                            (HarvestObject.state != u'COMPLETE'),
+                            (HarvestObject.state != u'ERROR'),
+                            (HarvestObject.state != u'STUCK')
```

Comment:
`STUCK` isn't a standard state, so let's not introduce it.

Reply:
All the _harvest_jobs_run_ function was added here to allow sending notifications for all job and with more info. 
If you agree I think we need to create an issue at upstream and fix it [there](https://github.com/ckan/ckanext-harvest/blob/94f0438c0ec6bfd7c85e64a3c15338a4d7800f2c/ckanext/harvest/logic/action/update.py#L568). 
It will be easy to do it, we can add a new config value to notify all jobs.
We may also improve the information within the email to meet GSA requirements.
I propose to delete this function here and fix this at upstream.